2.evaluate_models/log_reg_plates_3_3p_5_cp_norm_data/nbconverted/plate_3_3p_5_evaluate_log_reg.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Calculate evaluation metrics from training, validation, and testing data
+
+# In[1]:
+
+
+import pathlib
+from collections import defaultdict
+
+import pandas as pd
+from joblib import load
+from sklearn.metrics import confusion_matrix, f1_score, precision_recall_curve
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define paths
+
+# ### Input
+
+# In[3]:
+
+
+data_suf = "log_reg_cp_fs_data_plate_5"
+
+data_path = pathlib.Path(f"{root_dir}/1.train_models/log_reg_plates_3_3p_5_cp_norm_data/data")
+
+model_predf = pd.read_parquet(f"{data_path}/model_data_{data_suf}.parquet")
+evaldf = pd.read_parquet(f"{data_path}/model_data_log_reg_cp_fs_data_plate_5.parquet")
+le = load(f"{data_path}/label_encoder_log_reg_cp_fs_data_plate_5.joblib")
+
+
+# ### Outputs
+
+# In[4]:
+
+
+eval_path = pathlib.Path("model_eval_data")
+eval_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[5]:
+
+
+gene_column = "true_genotype"
+
+def down_sample_by_genotype(_df):
+    """
+    Parameters
+    ----------
+    _df: Pandas Dataframe
+        The data to be downsampled by the gene_column column.
+
+    Returns
+    -------
+        The data down-sampled by genotype.
+    """
+
+    min_gene = _df[gene_column].value_counts().min()
+    return (_df.groupby(gene_column, group_keys=False)
+            .apply(lambda x: x.sample(n=min_gene, random_state=0))
+            )
+
+
+# ## Calculate evaluation metrics
+
+# In[6]:
+
+
+# Define evaluation metrics
```

Comment:
Same as comment above, this might just be a different style of function writing but since you have these variables in your function, it might make sense to include them as parameters.

Reply:
See above for explanation